fendermint/vm/interpreter/src/fvm/state/query.rs

```diff
@@ -125,8 +103,13 @@ where
     }
 
     /// Get the state of an actor, if it exists.
-    pub fn actor_state(&self, addr: &Address) -> anyhow::Result<Option<(ActorID, ActorState)>> {
-        self.with_state_tree(|state_tree| {
+    pub fn actor_state(
+        &self,
+        use_cache: bool,
+        addr: &Address,
+    ) -> anyhow::Result<Option<(ActorID, ActorState)>> {
+        self.with_exec_state(use_cache, |exec_state| {
```

Comment:
I will probably let @aakoshh have the final call in this one because I may be missing something, so let's wait for his review, but do we really need `use_cache` as a public argument of `actor_state`? Isn't `use_cache` always `false` in this case? 
```suggestion
        self.with_exec_state(false, |exec_state| {
```

Reply:
I think it makes sense to offer it, the calls can now influence the result, and you might want to query it taking the last call into account.